shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+(function (global) {
+  'use strict';
+
+  var softKeyModule = (function () {
+
+    // Keys which are mapped to accesskey
+    const SK_KEYS_MAPPING = {
+      LSK: 'a',
+      CSK: 's',
+      RSK: 'd'
+    };
+
+    // Storing the reference to softkeys
+    var sk_keys = {
+      LSK: null,
+      CSK: null,
+      RSK: null
+    };
+
+    // Container created for SoftKey
+    var sk_container = null;
+
+    // Check if menu open
+    var sk_options_menu = {
+      open: false,
+      cancelKey: SK_KEYS_MAPPING.LSK,
+      cancelKeyText: 'Cancel',
+      cancelKeyl10nId: 'cancel'
+    };
+
+    // Observer for observing the DOM changes
+    var observer = null;
+
+    return {
+      init: init
+    };
+
+    /**
+     * Used to initialize the library
+     * @example
+     * softKeyModule.init();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function init() {
+
+      // Calling destroy to destroy previously initialized variables
+      destroy();
+
+      // Selecting the softkey DOM element
+      var softKey = document.querySelector('[role="softkey"]');
+      if (!softKey) {
+        // If null, return
+        return;
+      }
+
+      // Container will be created to handle options
+      // menu and soft keys
+      createSoftKeyContainer();
```

Comment:
It would probably be helpful to think of the softkey container as a separate module (or even a Web Component) that is responsible for displaying the soft key labels and handling the key events for the soft keys